src/arrayMethodSort.js

```
@@ -1,11 +1,34 @@
 'use strict';
+/* eslint-disable */
 
 /**
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  Array.prototype.sort2 = function (compareFunction) {
```

If a `compareFunction` is provided and is not a function, the method should throw a `TypeError`, matching the native `Array.prototype.sort` behavior (requirement 3.1). Currently, the code does not check for this and will silently use the default comparator if a non-function is passed. Consider adding a check at the start of the method:
```javascript
if (compareFunction !== undefined && typeof compareFunction !== 'function') {
  throw new TypeError('The comparison function must be either a function or undefined');
}
```